### ISSUE TITLE
Airship 15.0.1 MessageCenter - InboxMessageList API missing for read and delete messages

### ISSUE THREAD

**Issue by SwiftNativeDeveloper** [urbanairship/ios-library#293]
# Preliminary Info

### What Airship dependencies are you using?
Airship SPM 14.7.0 and Airship SPM 15.0.1

### What are the versions of any relevant development tools you are using?
Xcode 13.0 public

# Report

### What unexpected behavior are you seeing?
InboxMessageList API for marking messages read and deleted are not available.

### What is the expected behavior?
The API changes in version 15.0 of the library should not remove functionality from MessageCenter

### What are the steps to reproduce the unexpected behavior?

This code compiles against 14.7.0 SPM.

```

import SwiftUI
import AirshipMessageCenter

struct ContentView: View {
    var body: some View {
        NavigationView {
            List {
                Section(header: Text("MessageList")) {
                    ForEach(UAMessageCenter.shared().messageList.messages, id: \.self) { message in
                        VStack {
                            Text("Message: \(message.title)")
                            
                            HStack {
                                Button("Mark as read") {
                                    print("Reading message: \(message.messageID)")
                                    UAMessageCenter.shared().messageList.markMessagesRead([message]) {
                                        print("Read message: \(message.messageID)")
                                    }
                                }
                                
                                Button("Delete message") {
                                    print("Deleting message: \(message.messageID)")
                                    UAMessageCenter.shared().messageList.markMessagesDeleted([message]) {
                                        print("Deleted message: \(message.messageID)")
                                    }
                                }
                            }
                        }
                    }
                }
            }
            .navigationTitle("Message Center")
        }
        .navigationViewStyle(.stack)
    }
}

```

This code making the rename and singleton changes against 15.0.1 SPM does not compile. 

```

import SwiftUI
import AirshipMessageCenter

struct ContentView: View {
    var body: some View {
        NavigationView {
            List {
                Section(header: Text("MessageList")) {
                    ForEach(MessageCenter.shared.messageList.messages, id: \.self) { message in
                        VStack {
                            Text("Message: \(message.title)")
                            
                            HStack {
                                Button("Mark as read") {
                                    print("Reading message: \(message.messageID)")
                                    MessageCenter.shared.messageList.markMessagesRead([message]) {
                                        print("Read message: \(message.messageID)")
                                    }
                                }
                                
                                Button("Delete message") {
                                    print("Deleting message: \(message.messageID)")
                                    MessageCenter.shared.messageList.markMessagesDeleted([message]) {
                                        print("Deleted message: \(message.messageID)")
                                    }
                                }
                            }
                        }
                    }
                }
            }
            .navigationTitle("Message Center")
        }
        .navigationViewStyle(.stack)
    }
}

```

Xcode complains that the markMessagesRead and markMessagesDeleted don't exist.

![Screen Shot 2021-09-23 at 1 48 36 PM](https://user-images.githubusercontent.com/73653238/134566315-267310ee-e8d3-41cf-9930-49fb32a57086.png)

When I navigate the source, the APIs still exist in the objective-c. I don't know enough about the conversion of obj-c to SPM but something feels broken.

![Screen Shot 2021-09-23 at 1 50 09 PM](https://user-images.githubusercontent.com/73653238/134566465-4fa59d51-69b7-4e9a-a9b1-d15bc3ae660d.png)

**Comment by marc-scig** [comment]
Hi @SwiftNativeDeveloper,

That is quite strange. I'm able to reproduce the issue with a basic SPM app, but we really did not change anything about AirshipMessageCenter in this release aside from adding NS_SWIFT_NAME annotations to drop the UA prefix. SPM with obj-c works normally. I'll open a ticket to investigate this, but I'm afraid I don't have a good answer yet. Thanks for letting us know.

**Comment by SwiftNativeDeveloper** [comment]
Thanks @marc-scig for your quick reply! I do really appreciate the swift-style API changes that were made in this release.

**Comment by rlepinski** [comment]
@SwiftNativeDeveloper It looks like objective-c modules that reference types from our swift modules are not generating swift bindings when using SPM.  This seems like an SPM bug, but we are trying to work around it.  Swift to Swift modules are unaffected. Luckily it looks like we only expose a few core types in IAA and Message Center: Padding, Disposable, JsonPredicate, and VersionMatcher. 

I am going to move Padding and Disposable to our obj-c basement module (used for our migration to swift), which should resolve your issue. JsonPredicate and VersionMatcher are only exposed in a few classes that I doubt are used very often, so I am going to leave those as is for now. Anyone trying to access that from swift will have to use pods or xcframeworks for now, or wrap your usage in an obj-c class that you call through swift.

I hope to have an update this week.

**Comment by rlepinski** [comment]
Fixed in 16.0.0. Its a major release because we changed the pod import statement, which will not affect you.
